reaktive/src/commonMain/kotlin/com/badoo/reaktive/observable/SwitchIfEmpty.kt

```diff
@@ -0,0 +1,52 @@
+package com.badoo.reaktive.observable
+
+import com.badoo.reaktive.base.ErrorCallback
+import com.badoo.reaktive.disposable.Disposable
+import com.badoo.reaktive.disposable.DisposableWrapper
+import com.badoo.reaktive.utils.atomicreference.AtomicReference
+
+fun <T> Observable<T>.switchIfEmpty(otherObservable: Observable<T>): Observable<T> =
+    switchIfEmpty { otherObservable }
+
+fun <T> Observable<T>.switchIfEmpty(otherObservable: () -> Observable<T>): Observable<T> =
+    observableUnsafe { observer ->
+        val disposableWrapper = DisposableWrapper()
+        observer.onSubscribe(disposableWrapper)
+
+        subscribeSafe(
+            object : ObservableObserver<T>, ObservableCallbacks<T> by observer {
```

Comment:
You are overriding `onNext` and `onComplete` so you can replace `ObservableCallbacks<T>` with just `ErrorCallback`.